### PR TITLE
Add some more opentracing to sliding sync

### PR DESCRIPTION
**Pull request by erikjohnston**
This will make it easier to see what it is doing in jaeger.

### CODE REVIEW

File: synapse/rest/client/sync.py [hunk 1]
```diff
@@ -897,7 +897,23 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
         # by filter ID. For now, we will just prototype with always passing everything
         # in.
         body = parse_and_validate_json_object_from_request(request, SlidingSyncBody)
-        logger.info("Sliding sync request: %r", body)
+
+        # Tag and log useful data to differentiate requests.
+        set_tag("sliding_sync.conn_id", body.conn_id or "")
```
Comment (anoadragon453) [inline synapse/rest/client/sync.py:902]: How come we don't want these fields to be `None`? (`... or ""`)

File: synapse/handlers/sliding_sync.py [hunk 2]
```diff
@@ -1492,6 +1496,10 @@ async def get_room_sync_data(
             else:
                 assert_never(room_status.status)
 
+            log_kv({"sliding_sync.room_status": room_status})
+
+        log_kv({"sliding_sync.from_bound": from_bound, "sliding_sync.initial": initial})
```
Comment (anoadragon453) [inline synapse/handlers/sliding_sync.py:1501]: ...but `from_bound` here can be `None`?

**Review by anoadragon453 — APPROVED** [review]
lgtm, just one question.

File: synapse/handlers/sliding_sync.py (hunk 2 above)
Comment (erikjohnston) [reply synapse/handlers/sliding_sync.py:1501]: The logs will just stringify.

File: synapse/rest/client/sync.py (hunk 1 above)
Comment (erikjohnston) [reply synapse/rest/client/sync.py:902]: The `set_tag` does not like `None`, not entirely sure if that is an actual opentracing thing, or if that is a typing thing

**Pull request merged**